rflx/expression.py

```diff
@@ -40,26 +40,33 @@ class ProofResult(Enum):
 
 
 class Proof:
+    LOGIC = "QF_NIA"
+
     def __init__(self, expr: "Expr", facts: Optional[Sequence["Expr"]] = None):
         self.__expr = expr
         self.__facts = facts or []
         self.__result = ProofResult.UNSAT
+        self.__unknown_reason = "unknown"
 
-        solver = z3.Solver()
+        solver = z3.SolverFor(self.LOGIC)
         solver.add(self.__expr.z3expr())
         for f in self.__facts:
             solver.add(f.z3expr())
 
         self.__result = ProofResult(solver.check())
+        if self.__result == ProofResult.UNKNOWN:
+            self.__unknown_reason = solver.reason_unknown()
 
     @property
     def result(self) -> ProofResult:
         return self.__result
 
     @property
     def error(self) -> List[Tuple[str, Optional[Location]]]:
+        if self.__result == ProofResult.UNKNOWN:
+            return [(self.__unknown_reason, None)]
         assert self.__result == ProofResult.UNSAT
```

Comment:
What about returning an empty list in case of `SAT` instead of having the assert?

Reply:
You're not supposed to call error in case of `SAT`. I dislike the idea of hiding these kind of usage errors by returning an empty list, as they often hint another problem (e.g. you forgot to properly check the proof result).